app/routes/companyInterest/CompanyInterestListRoute.ts

```diff
@@ -40,25 +45,57 @@ const mapStateToProps = (state, props) => {
           })
         : 'Vis alle semestre',
   };
+  const selectedEventOption = {
+    value: eventValue ? eventValue : '',
+    label: eventValue
+      ? event_type_options
+          .filter((eventType) => eventType.value === eventValue)
+          .at(0).label
+      : 'Vis alle arrangementstyper',
+  };
   const companyInterestList = selectCompanyInterestList(
     state,
-    selectedOption.id
+    selectedSemesterOption.id,
+    selectedEventOption.value
   );
   const hasMore = state.companyInterest.hasMore;
   const fetching = state.companyInterest.fetching;
+  const event_string = selectedEventOption.value
+    ? `-${selectedEventOption.value}`
+    : '';
   return {
     semesters,
     companyInterestList,
     hasMore,
     fetching,
-    selectedOption,
+    selectedSemesterOption,
+    selectedEventOption,
+
+    exportSurvey: async (event?: string) => {
+      const blob = await fetch(
+        getCsvUrl(
+          selectedSemesterOption.year,
+          selectedSemesterOption.semester,
+          event
+        ),
+        {
+          headers: {
+            Authorization: `Bearer ${state.auth.token}`,
+          },
+        }
+      ).then((response) => response.blob());
+      return {
+        url: URL.createObjectURL(blob),
+        filename: `Company-interests-${selectedSemesterOption.year}-${selectedSemesterOption.semester}${event_string}.csv`,
+      };
```

Comment:
This should be a user action located in `CompanyInterestActions.ts`, using the `callAPI`. Then just map that dispatch to props.

Reply:
That is preferable, but `CompanyInterestActions.ts` doest have access to `selectedSemesterOption.year` and `selectedSemesterOption.semester`. Should I:
1. copy paste the code for selectedSemester in the actions file or
2. export selectedSemester or
3. keep the function in this file

Reply:
I relocated this function to the `CompanyInterestList.ts`